i3_resurrect/layout.py

```diff
@@ -33,34 +33,38 @@ def save(workspace, numeric, directory, profile, swallow_criteria):
         )
 
 
-def restore(workspace, numeric, directory, profile):
+def read(workspace, directory, profile):
     """
-    Restore an i3 workspace layout.
+    Read saved layout file.
     """
     filename = f'workspace_{workspace}_layout.json'
     if profile is not None:
         filename = f'{profile}_layout.json'
     layout_file = Path(directory) / filename
-
-    # Read saved layout file.
     layout = None
     try:
         layout = json.loads(layout_file.read_text())
-        if layout == {}:
-            return
     except FileNotFoundError:
         if profile is not None:
             util.eprint(f'Could not find saved layout for profile "{profile}"')
         else:
             util.eprint('Could not find saved layout for workspace '
                         f'"{workspace}"')
         sys.exit(1)
+    return layout
 
+
+def restore(workspace_name, profile, layout):
```

Comment:
Actually the `profile` argument can also be removed now because it is only used in the `except FileNotFoundError` block which you can also delete (see other comment).